cmd/konnectd/serve.go

```diff
@@ -36,52 +39,66 @@ import (
 	"stash.kopano.io/kc/konnect/version"
 )
 
-// Defaults.
-const (
-	defaultListenAddr           = "127.0.0.1:8777"
-	defaultIdentifierClientPath = "./identifier-webapp"
-	defaultSigningKeyID         = "default"
-	defaultSigningKeyBits       = 2048
-)
+var bootstrapConfig = &bootstrap.Config{}
 
 func commandServe() *cobra.Command {
 	serveCmd := &cobra.Command{
 		Use:   "serve <identity-manager> [...args]",
 		Short: "Start server and listen for requests",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) < 1 {
+				return fmt.Errorf("identity-manager argument missing, use one of kc, ldap, cookie, dummy")
+			}
+
+			bootstrapConfig.IdentityManager = args[0]
+			if bootstrapConfig.IdentityManager == "cookie" {
```

Comment:
Mhm, this identity manager dependent behavior here. I think it should move to the identity managers specific bootstrap file.

Reply:
Then the cookie bootstrap would need to know about args which bleeds the environment back in to the bootstrap logic?

There are more instances of this which I would like to get rid of in a separate PR

Reply:
OK :+1: